lib/vagrant-orchestrate/command/push.rb

```diff
@@ -56,7 +56,8 @@ def execute
           argv = parse_options(opts)
           return unless argv
 
-          guard_clean unless ENV["VAGRANT_ORCHESTRATE_NO_GUARD_CLEAN"]
+          guard_clean unless ENV["VAGRANT_ORCHESTRATE_NO_GUARD_CLEAN"] && \
+                             @env.vagrantfile.config.orchestrate.disable_commit_guard
```

Comment:
I changed tactics and updated to use a guard (early return) clause in the function, which should hopefully clarify the intent and simplify the mental overhead of processing multiple unless statements.